search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/TenantFilterTransformer.java

```diff
@@ -31,16 +29,16 @@ public TenantFilterTransformer(final IndexDescriptor indexDescriptor) {
   @Override
   public SearchQuery toSearchQuery(final TenantFilter filter) {
     return and(
-        term(TenantIndex.JOIN, IdentityJoinRelationshipType.TENANT.getType()),
         filter.key() == null ? null : term(KEY, filter.key()),
         filter.tenantId() == null ? null : term(TENANT_ID, filter.tenantId()),
         filter.name() == null ? null : term(NAME, filter.name()),
-        filter.memberKeys() == null
+        filter.tenantMemberType() == null
             ? null
-            : filter.memberKeys().isEmpty()
-                ? matchNone()
-                : hasChildQuery(
-                    IdentityJoinRelationshipType.MEMBER.getType(),
-                    longTerms(TenantIndex.MEMBER_KEY, filter.memberKeys())));
+            : term(TenantIndex.MEMBER_TYPE, filter.tenantMemberType().name()),
+        filter.joinParentId() == null
+            ? term(TenantIndex.JOIN, IdentityJoinRelationshipType.TENANT.getType())
+            : hasParentQuery(
+                IdentityJoinRelationshipType.TENANT.getType(),
+                term(TENANT_ID, filter.joinParentId())));
```

Comment:
It's not from this PR, but this always feels off to me: there are two separate entities (Tenant and TenantMember) that are squashed into one object. As a result, some of the filter configurations don't make sense (e.g. you can't combine a name and a parent id filter). Ideally, I'd split the entities completely and just let them share the index. Filters could be very clear then:
```java
record TenantFilter(String name, …) {…}
record TenantMemberFilter(TenantFilter parentFilter, EntityType memberType) {…}
```
This is really out of scope here though and should be a separate refactoring.

Reply:
I agree here in both point, firstly that it would be a cleaner approach and clearer but secondly that its not for this PR :)